### ISSUE TITLE
Profiles are hard to read

### ISSUE THREAD

**Issue by ana-pantilie** [kframework/kore#2675]
The (Profiteur) profiles are hard to read because of some laziness issues. In order to fix this, @ttuegel  suggested we rewrite the following:
- [whileReachability](https://github.com/kframework/kore/blob/2076934d030c1cd2b5a498104aed5481b841ac19/kore/src/Kore/Log/InfoReachability.hs#L39) so that all of the arguments are named and using do-notation
- [withDebugProven](https://github.com/kframework/kore/blob/5fddbb087dedd928b08414a23bca6c13dd672d59/kore/src/Kore/Reachability/Prove.hs#L591) should use do-notation

Actually, everything in [transitionRule'](https://github.com/kframework/kore/blob/5fddbb087dedd928b08414a23bca6c13dd672d59/kore/src/Kore/Reachability/Prove.hs#L399) should appear in the right order in the profiles.

Basically, every function in that file should look similar to [withWarnings](https://github.com/kframework/kore/blob/5fddbb087dedd928b08414a23bca6c13dd672d59/kore/src/Kore/Reachability/Prove.hs#L423).

**Comment by ana-pantilie** [comment]
@emarzion I was profiling `test/regression-wasm/test-wrc20-spec.sh` using the generated `.prof` file in Profiteur. Inside Profiteur, I was looking at alloc information.
